packages/next-international/src/app/middleware/index.ts

```diff
@@ -9,48 +9,51 @@ const DEFAULT_STRATEGY: NonNullable<I18nMiddlewareConfig<[]>['urlMappingStrategy
 
 export function createI18nMiddleware<const Locales extends readonly string[]>(config: I18nMiddlewareConfig<Locales>) {
   return function I18nMiddleware(request: NextRequest) {
-    const requestUrl = request.nextUrl.clone();
     const locale = localeFromRequest(config.locales, request, config.resolveLocaleFromRequest) ?? config.defaultLocale;
+    const nextUrl = request.nextUrl;
 
-    if (noLocalePrefix(config.locales, requestUrl.pathname)) {
-      requestUrl.pathname = `/${locale}${requestUrl.pathname}`;
+    // If the locale from the request is not an handled locale, then redirect to the same URL with the default locale
+    if (!hasHandledLocale(config.locales, nextUrl.pathname)) {
```

Comment:
Not a fan of the updated function name + the negation. Could we revert the changes?